internal/stores/hostdb.go

```diff
@@ -423,10 +435,12 @@ func (db *SQLStore) RecordInteractions(ctx context.Context, interactions []hostd
 				if isScan && host.LastScan > 0 && host.LastScan < interactionTime {
 					host.Uptime += time.Duration(interactionTime - host.LastScan)
 				}
+				host.RecentDowntime = 0
 			} else {
 				host.FailedInteractions++
 				if isScan && host.LastScan > 0 && host.LastScan < interactionTime {
 					host.Downtime += time.Duration(interactionTime - host.LastScan)
+					host.RecentDowntime += time.Duration(interactionTime - host.LastScan)
```

Comment:
This works but it has 2 downsides.

1. a host only needs to be scanned once to completely wipe it's downtime history. That's fine for now I'd say but something to keep in the back of our heads in case we want to update it to be a little less forgiving. e.g. subtract the uptime from the recent downtime or something like that to slowly decrease the downtime.
2. This is more problematic. If the max downtime is set to 1 week and renterd isn't running for more than 1 week, every host that is offline during the next scan is wiped. Even if it has only been offline for 5 minutes. So I think we should also keep a counter for how many scans the downtime is made up of and enforce a limit of at least multiple scans in addition to the downtime. e.g. 10.

Reply:
Yeah #2 is a good point. I'll make the change - good catch